packages/machine/src/protocol/install-virtual-app.ts

```diff
@@ -142,6 +143,25 @@ function createTarget(
   );
 }
 
+function addTarget(
+  context: Context,
+  initiatingAddress: string,
+  respondingAddress: string,
+  intermediaryAddress: string,
+  targetAppInstance: AppInstance
+) {
+  const key = virtualAppKey(
+    [initiatingAddress, respondingAddress],
+    intermediaryAddress
+  );
+  let sc = context.stateChannelsMap.get(key);
+  if (sc === undefined) {
```

Comment:
You can use the `.has` method instead of checking if `.get` returns `undefined`

Reply:
I think `.has` is not the best since it we need to do the `.get` in any case